core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -339,7 +339,7 @@ sealed abstract class IO[+E, +A] extends Serializable { self =>
    * because it's possible the fiber will be interrupted after acquisition but
    * before the finalizer is added.
    */
-  final def ensuring(finalizer: IO[Nothing, Any]): IO[E, A] =
+  final def ensuring(finalizer: IO[Nothing, _]): IO[E, A] =
```

Comment:
With all these changes, you can look for expressions like `io.void` and probably simplify to `io` most of the time (perhaps changing a `*>` into `<*` too)

Reply:
Good point, will take care of that.